test_cases/solidity/insufficient_gas_griefing/relayer_fixed/relayer_fixed.sol

```diff
@@ -0,0 +1,28 @@
+/*
+ * @source: https://consensys.github.io/smart-contract-best-practices/known_attacks/#insufficient-gas-griefing
+ * @author: ConsenSys Diligence
+ * Modified by Kaden Zipfel
+ */
+
+pragma solidity ^0.4.21;
+
+contract Relayer {
+    mapping (bytes => bool) executed;
+
+    function relay(bytes _data) public {
+        // replay protection; do not call the same transaction twice
+        require(executed[_data] == false);
+        executed[_data] = true;
+
+        address executor = new Executor();
+        executor.call(bytes4(keccak256("execute(bytes)")), _data);
+    }
+}
+
+// Contract called by Relayer
+contract Executor {
+    function execute(bytes _data, uint _gasLimit) public {
+        require(gasleft() >= _gasLimit);
```

Comment:
I think the right mitigation is probably debatable, but this is a pretty good one.
Is this what was done to address the issue in the audit on this: https://github.com/christianlundkvist/simple-multisig?